File: 프로그래머스/lv1/42889. 실패율/실패율.py
def solution(N, stages):
    answer = []
    
    stage_lst = []
    
    total = len(stages)
    for i in range(1, N + 1):
        if total == 0:
            stage_lst.append([0, i])
        else:
            cnt = stages.count(i)
            fail = cnt / total
            stage_lst.append([fail, i])
            total -= cnt
    
    stage_lst.sort(key = lambda x : x[0] , reverse = True)
    for s in stage_lst:
        answer.append(s[1])
    
    return answer

File: 프로그래머스/1/42889. 실패율/실패율.js
function solution(N, stages) {
    var answer = [];
    
    for (let i = 1; i < N + 1; i++) {        
        let fail = stages.filter(e => e === i).length / stages.length
        answer.push([i, fail])
        stages = stages.filter(e => e !== i)
    }
        
    return [...answer.sort((a, b) => b[1] - a[1]).map(stage => stage[0])];
}